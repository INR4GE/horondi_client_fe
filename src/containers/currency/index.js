import Language from './currency';

export default Language;
