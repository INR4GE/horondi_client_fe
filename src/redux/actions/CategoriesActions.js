const categoriesLoaded = (newCategories) => ({
  type: 'CATEGORIES_LOADED',
  payload: newCategories
});

const categoriesRequested = () => ({
  type: 'CATEGORIES_REQUESTED'
});

const categoryLoaded = (newCategory) => ({
  type: 'CATEGORY_LOADED',
  payload: newCategory
});

export { categoriesLoaded, categoryLoaded, categoriesRequested };
