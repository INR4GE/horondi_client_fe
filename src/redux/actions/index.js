import {
  categoriesLoaded,
  categoryLoaded,
  categoriesRequested
} from './CategoriesActions';

export {
  categoriesLoaded,
  categoryLoaded,
  categoriesRequested
};
