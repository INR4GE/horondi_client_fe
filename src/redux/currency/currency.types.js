const CHANGE_CURRENCY = 'CHANGE_CURRENCY';

export { CHANGE_CURRENCY };
