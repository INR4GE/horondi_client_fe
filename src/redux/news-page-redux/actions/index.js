import { storeSetNews, wachNewsLoad } from './news-actions/News-actions';

export { storeSetNews, wachNewsLoad };
