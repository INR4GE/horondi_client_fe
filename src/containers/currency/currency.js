import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { MenuItem } from '@material-ui/core';
import {
  setToLocalStorage,
  getFromLocalStorage
} from '../../services/local-storage.service';
import { changeCurrency } from '../../redux/currency/currency.actions';
import { CURRENCIES_LIST } from '../../translations/currency.translation';
import Dropdown from '../../components/dropdown';
import useStyles from './currency.styles';

const currencyInLocalStorage = getFromLocalStorage('currency') || 0;

const Currency = () => {
  const styles = useStyles();
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(changeCurrency(currencyInLocalStorage));
  }, [dispatch]);

  const handleChange = (e) => {
    const targetValue = e.target.value;
    if (targetValue !== undefined) {
      setToLocalStorage('currency', targetValue);
      dispatch(changeCurrency(targetValue));
    }
  };
  const mappedCurrencies = CURRENCIES_LIST.map(({ currency, value }) => (
    <MenuItem id={`currency${value + 1}`} key={value} value={value}>
      {currency === 'UAH' ? '\u20b4' : '\u0024'}
    </MenuItem>
  ));
  return (
    <div id='currency'>
      <Dropdown
        styles={styles}
        mappedItems={mappedCurrencies}
        handler={handleChange}
        defaultValue={currencyInLocalStorage}
      />
    </div>
  );
};

export default Currency;
