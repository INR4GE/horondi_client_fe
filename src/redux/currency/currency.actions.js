import { CHANGE_CURRENCY } from './currency.types';

const changeCurrency = (value) => ({
  type: CHANGE_CURRENCY,
  payload: value
});

export { changeCurrency };
