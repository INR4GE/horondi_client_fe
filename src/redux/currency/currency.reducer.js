import { CHANGE_CURRENCY } from './currency.types';

export const initialState = {
  currency: 0
};

const currencyReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case CHANGE_CURRENCY:
    return {
      ...state,
      currency: action.payload
    };

  default: {
    return state;
  }
  }
};

export default currencyReducer;
