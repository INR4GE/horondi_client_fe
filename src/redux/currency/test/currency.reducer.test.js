/* eslint-disable no-undef */
import currencyReducer, { initialState } from '../currency.reducer';
import { changeCurrency } from '../currency.actions';
import { CHANGE_CURRENCY } from '../currency.types';

describe('test currency reducer', () => {
  test('should return default state', () => {
    expect(currencyReducer()).toMatchSnapshot();
    expect(currencyReducer()).toEqual({
      currency: 0
    });
  });

  test('should change currency', () => {
    expect(currencyReducer(initialState, changeCurrency(1))).toMatchSnapshot();
    expect(currencyReducer(initialState, changeCurrency(1))).toEqual({
      currency: 1
    });
  });
});
