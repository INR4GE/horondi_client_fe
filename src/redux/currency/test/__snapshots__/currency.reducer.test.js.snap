// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test currency reducer should change currency 1`] = `
Object {
  "currency": 1,
}
`;

exports[`test currency reducer should return default state 1`] = `
Object {
  "currency": 0,
}
`;
